STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm_blob_ustream_interface.c

```diff
@@ -314,10 +310,6 @@ AZ_NODISCARD az_result az_blob_create_ustream_from_blob(
     // grab blob package size for ustream init
     uint32_t package_size
         = (uint32_t)blob_http_cb->_internal.http_client.nx_web_http_client_total_receive_bytes;
-    if ((package_size & 0x07FF) != 0x000)
-    {
-      package_size = (package_size & 0xFFFFF800) + 0x0800;
-    }
```

Comment:
Is there no need to round to the nearest 2K anymore?

Reply:
1. This is particular of the consumer of this stream. You cannot change the size of a stream to fit the needs of the consumer of it, if the consumer needs to resize it, the consumer shall do that by its own. 

2. The Flash APIs requires the real number of bytes that will be written. The eraser API will handle the block size, so you don't need to do that.

3. If you look at the API write in Flash, it relies on the right size of the file to write the last bytes, in the case that the size is not aligned to 8 bytes, change the size here was causing a problem with the last bytes.